airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobCreator.java

```diff
@@ -55,7 +60,19 @@ public DefaultJobCreator(final JobPersistence jobPersistence,
         .withOperationSequence(standardSyncOperations)
         .withConfiguredAirbyteCatalog(standardSync.getCatalog())
         .withState(null)
-        .withResourceRequirements(getJobResourceRequirements(standardSync));
+        .withResourceRequirements(ResourceRequirementsUtils.mergeResourceRequirements(
+            standardSync.getResourceRequirements(),
+            workerResourceRequirements))
+        .withSourceResourceRequirements(ResourceRequirementsUtils.getResourceRequirements(
+            standardSync.getResourceRequirements(),
+            sourceResourceReqs,
+            workerResourceRequirements,
+            JobType.SYNC))
+        .withDestinationResourceRequirements(ResourceRequirementsUtils.getResourceRequirements(
+            standardSync.getResourceRequirements(),
+            destinationResourceReqs,
+            workerResourceRequirements,
+            JobType.SYNC));
```

Comment:
This is the main change being made. Here we set the source and destination resource requirements according to the hierarchy order. The other changes in this PR are to support setting this on the job sync config, and then using these resource requirements to actually create the source/dest pods